tests/_compilertestrunner.livecodescript

```diff
@@ -0,0 +1,637 @@
+            put offset("%{", tLine, tPositionSkip) into tPositionOffset
+            if tPositionOffset is 0 then
+               put char tPositionSkip to -1 of tLine after tCode
+               exit repeat
+            end if
+            add tPositionSkip to tPositionOffset
+
+            put char tPositionSkip to tPositionOffset - 1 of tLine after tCode
+
+            -- Look for the end brace, it is an error if it is not there
+            local tPositionEndOffset
+            put offset("}", tLine, tPositionOffset) into tPositionEndOffset
+            if tPositionEndOffset is 0 then
+               reportCompilerTestError pScriptFile, tLineNumber, "unterminated named position"
+            end if
+            add tPositionOffset to tPositionEndOffset
+
+            -- Check that the name is not empty
+            local tPositionName
+            put char tPositionOffset + 2 to tPositionEndOffset - 1 of tLine into tPositionName
+            if tPositionName is empty then
+               reportCompilerTestError pScriptFile, tLineNumber, "empty named position"
+            end if
+
+            -- Check that the position hasn't already been defined
+            if tPositions[tPositionName] is not empty then
+               reportCompilerTestError pScriptFile, tLineNumber, "named position already defined"
+            end if
+
+            reportCompilerTestDiag format("found position '%s' at line %d", tPositionName, tLineNumber)
+            put tLineNumber - tTestLineNumber + 1 into tPositions[tPositionName]
+
+            put tPositionEndOffset + 1 into tPositionSkip
+         end repeat
+
+         -- Add the newline to the code
+         put return after tCode
+      else if tState is "assertions" then
+         -- We only allow directives inside assertion blocks
+         if word 1 to -1 of tLine is not empty then
+            reportCompilerTestError pScriptFile, tLineNumber, "junk inside assertion clause"
+         end if
+      end if
+   end repeat
+
+   if tName is empty then
+      reportCompilerTestError pScriptFile, tLineNumber, format("test '%s' not found in file", pTest)
+   end if
+
+   -- We now have the code for the test, the expectation and a list of
+   -- assertions.
+
+   put tName into rCompilerTest["name"]
+   put tCode into rCompilerTest["code"]
+   put tPositions into rCompilerTest["positions"]
+   put tExpectation into rCompilerTest["expectation"]
+   put tAssertions into rCompilerTest["assertions"]
+end processCompilerTest
+
+private command reportCompilerTestError pFile, pLine, pMessage
+   throw format("'%s', line %d: %s", pFile, pLine, pMessage)
+end reportCompilerTestError
+
+private command reportCompilerTestDiag pMessage
+   repeat for each line tLine in pMessage
+      write "DIAG:" && tLine & return to stderr
+   end repeat
+end reportCompilerTestDiag
+
+----------------------------------------------------------------
+-- Logging helpers
+----------------------------------------------------------------
```

Comment:
These are mostly specific to the compilerrunner at the moment - I've made it so that you can set the LCC_VERBOSE env var to 1 to emit all the info about what it finds in the compiler tests.